### PR TITLE
Correctly translate Views v2 URLs when using WPML

### PR DESCRIPTION
**Pull request by lucatume**
Issue: https://moderntribe.atlassian.net/browse/TEC-3608 

This PR restores and fixes the handling of translation for Views v2
URLs when the plugin is used together with WPML.`

[Screencast showing WPML support on TEC](https://drive.google.com/open?id=1HytQB0TtTOnvjgYWMmPdn5PmoMe-_SEq&authuser=luca%40tri.be&usp=drive_fs)
[Screencast showing vanilla EN site working correctly](https://drive.google.com/open?id=1I4RvB4LlD8YYwidtdZsjT0u9GNmGkUl3&authuser=luca%40tri.be&usp=drive_fs)
[Screencast showing vanilla non-EN (FR) site working correctly](https://drive.google.com/open?id=1I7YraTgX-tsTlYrS3Suq5aaVuxeiA1vB&authuser=luca%40tri.be&usp=drive_fs)

This PR main changes are:

1. Support and use case-insensitive URLs to build, and parse, localized slugs.
2. Support and use up-to-date translation information to build localized rewrite rules. Day view URLs would be incorrectly built, and parsed, due to our code (from TEC and PRO) and WordPress providing different translations for it. Part of this fix consists in checking whether a localized base already exists using transliteration.

The code changed in this PR is covered by pre-existing tests. So much so that not pulling the correct version of Common will break them. Actually, I found all the places I was breaking things exactly this way.

Based on https://github.com/moderntribe/tribe-common/pull/1488

### CODE REVIEW

File: src/Tribe/Views/V2/View.php [hunk 1]
```diff
@@ -1533,13 +1533,12 @@ protected function filter_repository_args( array $repository_args, Context $cont
...
-					home_url(),
-					'',
-					$this->rewrite->get_clean_url( (string) $this->get_url() ) ),
-				'/'
-			);
+		$plain_url   = (string) $this->get_url();
+		$clean_url   = $this->rewrite->get_clean_url( $plain_url );
+		$url_frags   = wp_parse_url( $clean_url );
+		$request_uri = '/' . ( isset( $url_frags['path'] ) ? trim( $url_frags['path'], '/' ) . '/' : '' )
+		               . ( isset( $url_frags['query'] ) ? '?' . $url_frags['query'] : '' )
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php:1540]: * :warning: Found precision alignment of 3 spaces.
* :no_entry_sign: Tabs must be used to indent lines; spaces are not allowed

<sub>hash: f96de5b40b3f5c001422ed5664e7054d</sub>

File: src/Tribe/Views/V2/View.php [hunk 2]
```diff
@@ -1533,13 +1533,12 @@ protected function filter_repository_args( array $repository_args, Context $cont
...
+		               . ( isset( $url_frags['fragment'] ) ? '#' . $url_frags['fragment'] : '' );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php:1541]: * :warning: Found precision alignment of 3 spaces.
* :no_entry_sign: Tabs must be used to indent lines; spaces are not allowed

<sub>hash: 4ef5ce52506906cb338c02a2009acbe9</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 2 error(s)
:warning: 2 warning(s)

File: src/Tribe/Integrations/WPML/Language_Switcher.php [hunk 3]
```diff
@@ -44,7 +44,7 @@ public function filter_icl_ls_languages( array $languages = array() ) {
 		$root_folder = parse_url( home_url(), PHP_URL_PATH );
 		$request_uri = $_SERVER['REQUEST_URI'];
 
-		if ( ! empty( $root_folder ) ) {
+		if ( $root_folder !== '/' ) {
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/Language_Switcher.php:47]: :no_entry_sign: Use Yoda Condition checks, you must.

<sub>hash: fd25b2a7c221b94eaad42957ea9526f7</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 3 error(s)
:warning: 2 warning(s)

File: src/Tribe/Views/V2/View.php [hunk 4]
```diff
@@ -1533,13 +1535,12 @@ protected function filter_repository_args( array $repository_args, Context $cont
...
-					home_url(),
-					'',
-					$this->rewrite->get_clean_url( (string) $this->get_url() ) ),
-				'/'
-			);
+		$plain_url   = (string) $this->get_url();
+		$clean_url   = $this->rewrite->get_clean_url( $plain_url );
+		$url_frags   = wp_parse_url( $clean_url );
+		$request_uri = '/' . ( isset( $url_frags['path'] ) ? trim( $url_frags['path'], '/' ) . '/' : '' )
+					   . ( isset( $url_frags['query'] ) ? '?' . $url_frags['query'] : '' )
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php:1542]: :warning: Found precision alignment of 3 spaces.

<sub>hash: c57e5ed91274ef6453330eb50615603d</sub>

File: src/Tribe/Views/V2/View.php [hunk 5]
```diff
@@ -1533,13 +1535,12 @@ protected function filter_repository_args( array $repository_args, Context $cont
...
+					   . ( isset( $url_frags['fragment'] ) ? '#' . $url_frags['fragment'] : '' );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php:1543]: :warning: Found precision alignment of 3 spaces.

<sub>hash: 9978d9ae3f1effa53030a4de2b3353c5</sub>

**Review by tr1b0t — COMMENTED** [review]
:warning: 2 warning(s)

File: src/Tribe/Views/V2/Url.php [hunk 6]
```diff
@@ -285,6 +285,16 @@ public static function from_url_and_params( $url = null, array $params = [] ) {
 			$url = home_url( add_query_arg( [] ) );
 		}
 
+		/*
+		 * Depending on the source of the request the encoded chars might be in uppercase (`%C3`) or lowercase (`%c2j`) format.
+		 * Ensure the format of the encoded chars is consistently lowercase.
+		 */
+		$lowercase_encoded_url = preg_replace_callback( '/%[A-F0-9]{2}/', static function ( array $matches ) {
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Url.php:292]: * :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call

<sub>hash: f7b9896270f224e0b6d00b00c6b37eef</sub>

File: src/Tribe/Views/V2/Url.php [hunk 7]
```diff
@@ -285,6 +285,16 @@ public static function from_url_and_params( $url = null, array $params = [] ) {
...
+			return strtolower( reset( $matches ) );
+		}, $url );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Url.php:294]: * :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: 8e6e5e73e0eeb82f8eba36ba0a7d54ad</sub>

File: src/Tribe/Views/V2/View.php [hunk 8]
```diff
@@ -258,8 +258,10 @@ public static function make_for_rest( \WP_REST_Request $request ) {
 			$params['prev_url'] = untrailingslashit( $params['prev_url'] );
 		}
 
-		$slug       = Arr::get( $params, 'view', false );
-		$url_object = Url::from_url_and_params( Arr::get( $params, 'url' ), $params );
+		$slug = Arr::get( $params, 'view', false );
+		// Convert the URL to lowercase to make sure the rewrite rules, all lowercase, will match it.
+		$url  = Arr::get( $params, 'url' );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php:263]: :warning: Equals sign not aligned with surrounding assignments; expected 8 spaces but found 2 spaces

<sub>hash: 71046e868662a64407e8f8c908cd6729</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 4 error(s)
:warning: 1 warning(s)

**Review by Camwyn — DISMISSED** [review]

**Pull request merged**